apps/arena/lib/arena/game/player.ex

```diff
@@ -261,27 +262,59 @@ defmodule Arena.Game.Player do
     player.aditional_info.inventory != nil
   end
 
-  def use_item(player, game_state) do
+  def use_item(player, game_state, game_config) do
     case player.aditional_info.inventory do
       nil ->
         game_state
 
       item ->
-        player =
-          Enum.reduce(item.effects, player, &apply_effect/2)
-          |> put_in([:aditional_info, :inventory], nil)
-
-        put_in(game_state, [:players, player.id], player)
+        Enum.reduce(item.effects, game_state, fn effect_name, game_state_acc ->
+          effect = Enum.find(game_config.effects, fn %{name: name} -> name == effect_name end)
+          Effect.put_effect(game_state_acc, player.id, player.id, effect)
+        end)
+        |> put_in([:players, player.id, :aditional_info, :inventory], nil)
     end
   end
 
-  def remove_damage_immunity(player) do
-    put_in(player, [:aditional_info, :damage_immunity], false)
-  end
-
   def invisible?(player) do
     get_in(player, [:aditional_info, :effects])
-    |> Enum.any?(fn {_, effect} -> effect.name == "invisible" end)
+    |> Enum.any?(fn effect ->
+      Enum.any?(effect.effect_mechanics, fn {mechanic, _} -> mechanic == :invisible end)
+    end)
+  end
+
+  def remove_expired_effects(player) do
+    now = System.monotonic_time(:millisecond)
+
+    effects =
+      player.aditional_info.effects
+      |> Enum.filter(fn effect -> effect.expires_at > now end)
+
+    put_in(player, [:aditional_info, :effects], effects)
+  end
+
+  def remove_effects_on_action(player) do
+    effects =
+      player.aditional_info.effects
+      |> Enum.reject(fn effect ->
+        effect.remove_on_action and
+          Enum.any?(player.aditional_info.current_actions, fn action -> action.action != :MOVING end)
+      end)
+
+    put_in(player, [:aditional_info, :effects], effects)
+  end
+
+  def reset_effects(player, game_config) do
+    character = Enum.find(game_config.characters, fn %{name: name} -> name == player.aditional_info.character_name end)
+
+    player =
+      player
+      |> put_in([:speed], character.base_speed)
+      |> put_in([:aditional_info, :stamina_interval], character.stamina_interval)
```

Comment:
maybe we could avoid this character fetch if we add the base_stats to the player and have them separated from the altereded ones, e.g. having base some variable called :base_speed and to reset it you just do `|> put_in([:speed], player.aditional_info.base_speed)`

Reply:
Changed in https://github.com/lambdaclass/mirra_backend/pull/425/commits/5fc4e9270b784b3237f16c2d601cf18b978b1e10